src/ray/gcs/gcs_server/gcs_publisher_manager.h

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <memory>
+#include "ray/common/asio/periodical_runner.h"
+#include "ray/pubsub/publisher.h"
+#include "ray/rpc/gcs_server/gcs_rpc_server.h"
+
+namespace ray {
+namespace gcs {
+
+/// This implementation class of `PublisherHandler`
+/// All non-handler public methods are threadsafe.
+class GcsPublisherManager : public rpc::PublisherHandler {
```

Comment:
Why not call it *Handler since it's just a rpc handler?

Reply:
I think the convention is Manager since I suppose it manages the lifecycle of the pubsub module? For example, `GcsHeartbeatManager` also doesn't really manage anything.